homeassistant/components/zha/number.py

```diff
@@ -553,6 +559,16 @@ def __init__(
             self._attr_native_min_value: float = self._channel.min_mireds
             self._attr_native_max_value: float = self._channel.max_mireds
 
+    def _convert_to_state_value(
+        self, value: float, method: Callable[[float, int], float]
+    ) -> float:
+        """Convert a value in the number's native unit to the configured unit."""
+        return color_temperature_mired_to_kelvin(value)
+
+    def convert_to_native_value(self, value: float) -> float:
+        """Convert a value to the number's native unit."""
```

Comment:
```suggestion
        """Convert a value to the number's native unit.

        This is overridden because NumberEntity does not know about converting between
        K and mireds, and the user can't choose the preferred color temp unit of a number..
        """
```